src/dram_controller.cc

```diff
@@ -552,21 +552,46 @@ bool MEMORY_CONTROLLER::add_wq(const request_type& packet)
   return false;
 }
 
+unsigned long DRAM_ADDRESS_MAPPING::swizzle_bits(champsim::address address, unsigned long segment_size, unsigned long segment_offset, unsigned long field, unsigned long field_bits) const {
```

Comment:
```suggestion
unsigned long DRAM_ADDRESS_MAPPING::swizzle_bits(champsim::address address, unsigned long segment_size, champsim::data::bits segment_offset, unsigned long field, unsigned long field_bits) const {
```
This value is only ever used as this type, so you should make it part of the interface.